src/dmd/parse.d

```diff
@@ -6751,6 +6751,22 @@ final class Parser(AST) : Lexer
             error(e.loc, "`%s` must be surrounded by parentheses when next to operator `%s`", e.toChars(), Token.toChars(value));
     }
 
+    private void checkParensNot(AST.Expression left, AST.Expression right, AST.Expression exp)
+    {
+        if (left.op == TOK.not && !left.parens)
+        {
+            import dmd.expression : UnaExp;
+
+            error(exp.loc, "`%s`: Boolean operator `%s` has higher precedence than bitwise operator `%s`.",
```

Comment:
That should probably be `exp.error` so you don't have to repeat the `exp.toChars()` or pass `loc`